### PR TITLE
Update the generator docs

### PR DESCRIPTION
**Pull request by dhaiducek**
- Add `generate*Placement` fields
- Add `policySetDefaults`
- Add `copyPolicyMetadata`
- Update Placement fields for LabelSelectors

ref: https://issues.redhat.com/browse/ACM-3011

### CODE REVIEW

**Review by mprahl — APPROVED** [review]

**Comment by dockerymick** [comment]
@dhaiducek thanks for creating this! I will review it today

**Comment by dockerymick** [comment]
@dhaiducek Thanks again for starting this PR. I have made some updates to the descriptions, along with adding the What's new entry. Will you review when you have an opportunity? Also getting requesting peer review from a writer

**Review by dockerymick — COMMENTED** [review]
Based on my updates, I approve this PR. Awaiting peer review from another writer

File: governance/policy_generator.adoc [hunk 1]
```diff
@@ -188,6 +188,10 @@ Note that all the fields in the `policyDefaults` section except for `namespace`
 | Optional
 | Key-value pairs of annotations to set on generated configuration policies. For example, you can disable policy templates by defining the following parameter: `{"policy.open-cluster-management.io/disable-templates": "true"}`. The default value is empty (`{}`).
 
+| `policyDefaults.copyPolicyMetadata`
+| Optional
+| Copies the labels and annotations for all policies and add to a replica policy. Set to `true` by default. If set to `false`, only the policy framework specific policy labels and annotations are copied to the replicated policy.
```
Comment (cadawson) [inline governance/policy_generator.adoc]: This one is confusing me a bit:
Copies the labels and annotations for all policies and add them to a replica policy. (?)
Copies the labels and annotations for all policies and adds them to a replica policy.(?)

File: governance/policy_generator.adoc [hunk 2]
```diff
@@ -302,27 +310,39 @@ Note that all the fields in the `policyDefaults` section except for `namespace`
 
 | `policyDefaults.placement.placementPath`
 | Optional
-| To reuse an existing placement, specify the path here relative to the `kustomization.yaml` file. If provided, this placement rule is used by all policies by default. See `clusterSelectors` to generate a new `Placement`.
+| To reuse an existing placement, specify the path here relative to the `kustomization.yaml` file. If provided, this placement rule is used by all policies by default. See `_clusterSelectors_` to generate a new `Placement`.
```
Comment (cadawson) [inline governance/policy_generator.adoc]:
```suggestion
| To reuse an existing placement, specify the path relative to the location of the `kustomization.yaml` file. If provided, this placement rule is used by all policies by default. See `_clusterSelectors_` to generate a new `Placement`.
```

File: governance/policy_generator.adoc [hunk 3]
```diff
@@ -302,27 +310,39 @@ Note that all the fields in the `policyDefaults` section except for `namespace`
...
 
-| `policyDefaults.placement.clusterSelectors`
+| `policyDefaults.placement.clusterSelector`
 | Optional
-| Specify a placement by defining a cluster selector in the following format, `key:value`. See `placementPath` to specify an existing file.
+| Specify a placement by defining a cluster selector using either `key:value`, or providing `matchExpressions` and or `matchLabels` keys with appropriate values. See `_placementPath_` to specify an existing file.
```
Comment (cadawson) [inline governance/policy_generator.adoc]:
```suggestion
| Specify a placement by defining a cluster selector using either `key:value`, or by providing `matchExpressions`, `matchLabels` keys, or both, with appropriate values. See `_placementPath_` to specify an existing file.
```

File: governance/policy_generator.adoc [hunk 4]
```diff
@@ -302,27 +310,39 @@ Note that all the fields in the `policyDefaults` section except for `namespace`
...
 
 | `policyDefaults.placement.placementRuleName`
 | Optional
 | To use a placement rule that already exists on the cluster, specify its name here. A `PlacementRule` is not created, but a `PlacementBinding` binds the policy to this `PlacementRule`.
 
 | `policyDefaults.placement.placementRulePath`
 | Optional
-| To reuse an existing placement rule, specify the path here relative to the `kustomization.yaml` file. If provided, this placement rule is used by all policies by default. See `labelSelector` to generate a new `PlacementRule`.
+| To reuse an existing placement rule, specify the path here relative to the `kustomization.yaml` file. If provided, this placement rule is used by all policies by default. See `_labelSelector_` to generate a new `PlacementRule`.
```
Comment (cadawson) [inline governance/policy_generator.adoc]:
```suggestion
| To reuse an existing placement rule, specify the path relative to the location of the `kustomization.yaml` file. If provided, this placement rule is used by all policies by default. See `_labelSelector_` to generate a new `PlacementRule`.
```

File: governance/policy_generator.adoc [hunk 5]
```diff
@@ -302,27 +310,39 @@ Note that all the fields in the `policyDefaults` section except for `namespace`
...
 
 | `policyDefaults.placement.labelSelector`
 | Optional
-| Specify a placement rule by defining a cluster selector in the following format, `key:value`. See `placementRulePath` to specify an existing file.
+| Specify a placement rule by defining a cluster selector using either `key:value`, or providing a `matchExpressions` and or `matchLabels` keys with appropriate values. See `_placementRulePath_` to specify an existing file.
```
Comment (cadawson) [inline governance/policy_generator.adoc]:
```suggestion
| Specify a placement rule by defining a cluster selector using either `key:value`, or providing a `matchExpressions`, `matchLabels` keys, or both, with appropriate values. See `_placementRulePath_` to specify an existing file.
```

File: release_notes/whats_new.adoc [hunk 6]
```diff
@@ -47,21 +47,9 @@ For other Application topics, see link:../applications/app_management_overview.a
...
-
-* Use the `toLiteral` function to remove any quotation marks around a template string after it is processed. See link:../governance/custom_template_functions.adoc#toliteral-function[_toLiteral_ function] for more details.
-
-* You can manage policy definitions with OpenShift GitOps (ArgoCD). See link:../governance/manage_policy_def.adoc#manage-policy-definitions[Managing policy definitions with OpenShift GitOps (ArgoCD)].
-
-* You now receive the status events for a policy patching an object in the _History_ column. For more information, see link:../governance/manage_policy_intro.adoc#grc-view[Governance page].
+//clearing what's new for 2.8 | MJ | 03/12/2023
+//2.8 ACM-3011 | MJ | 03/12/2023
+* You can now configure the Policy Generator to add a copy of meta data to replicas of your policies and policy sets, along with generate placement manifests for your policies and policy sets. See link:../governance/policy_generator.adoc#policy-gen-yaml-table[Policy Generator configuration reference table] for more details.
```
Comment (cadawson) [inline release_notes/whats_new.adoc]:
```suggestion
* You can now configure the Policy Generator to add a copy of metadata to replicas of your policies and policy sets, along with generate placement manifests for your policies and policy sets. See link:../governance/policy_generator.adoc#policy-gen-yaml-table[Policy Generator configuration reference table] for more details.
```

**Review by cadawson — CHANGES_REQUESTED** [review]
A few smaller suggestions. I don't need to review it again. Thanks!

File: governance/policy_generator.adoc (hunk 1 above)
Comment (dockerymick) [reply governance/policy_generator.adoc]: @cadawson thanks for your review, feedback, and suggestions!

**Comment by dockerymick** [comment]
@dhaiducek I have updated the PR based on the peer review from Chris. When you have an opportunity, will you rereview the PR?

File: governance/policy_generator.adoc [hunk 7]
```diff
@@ -302,27 +310,39 @@ Note that all the fields in the `policyDefaults` section except for `namespace`
 
 | `policyDefaults.placement.placementPath`
 | Optional
-| To reuse an existing placement, specify the path here relative to the `kustomization.yaml` file. If provided, this placement rule is used by all policies by default. See `clusterSelectors` to generate a new `Placement`.
+| To reuse an existing placement, specify the path relative to the location of the `kustomization.yaml` file. If provided, this placement rule is used by all policies by default. See `_clusterSelectors_` to generate a new `Placement`.
```
Comment (dhaiducek) [inline governance/policy_generator.adoc]:
```suggestion
| To reuse an existing placement, specify the path relative to the location of the `kustomization.yaml` file. If provided, this placement rule is used by all policies by default. See `_labelSelector_` to generate a new `Placement`.
```

File: governance/policy_generator.adoc [hunk 8]
```diff
@@ -302,27 +310,39 @@ Note that all the fields in the `policyDefaults` section except for `namespace`
...
 
-| `policyDefaults.placement.clusterSelectors`
+| `policyDefaults.placement.clusterSelector`
 | Optional
-| Specify a placement by defining a cluster selector in the following format, `key:value`. See `placementPath` to specify an existing file.
+| Specify a placement by defining a cluster selector using either `key:value`, or by providing `matchExpressions`, `matchLabels` keys, or both, with appropriate values. See `_placementPath_` to specify an existing file.
```
Comment (dhaiducek) [inline governance/policy_generator.adoc]:
```suggestion
| Specify a placement by defining a cluster selector using either `key:value`, or by providing `matchExpressions`, `matchLabels`, or both, with appropriate values. See `_placementRulePath_` to specify an existing file.
```

File: governance/policy_generator.adoc [hunk 9]
```diff
@@ -302,27 +310,39 @@ Note that all the fields in the `policyDefaults` section except for `namespace`
...
 
 | `policyDefaults.placement.placementRuleName`
 | Optional
 | To use a placement rule that already exists on the cluster, specify its name here. A `PlacementRule` is not created, but a `PlacementBinding` binds the policy to this `PlacementRule`.
 
 | `policyDefaults.placement.placementRulePath`
 | Optional
-| To reuse an existing placement rule, specify the path here relative to the `kustomization.yaml` file. If provided, this placement rule is used by all policies by default. See `labelSelector` to generate a new `PlacementRule`.
+| To reuse an existing placement rule, specify the path relative to the location of the `kustomization.yaml` file. If provided, this placement rule is used by all policies by default. See `_labelSelector_` to generate a new `PlacementRule`.
```
Comment (dhaiducek) [inline governance/policy_generator.adoc]:
```suggestion
| To reuse an existing placement rule, specify the path relative to the location of the `kustomization.yaml` file. If provided, this placement rule is used by all policies by default. See `_clusterSelector_` to generate a new `PlacementRule`.
```

File: governance/policy_generator.adoc [hunk 10]
```diff
@@ -302,27 +310,39 @@ Note that all the fields in the `policyDefaults` section except for `namespace`
...
 
 | `policyDefaults.placement.labelSelector`
 | Optional
-| Specify a placement rule by defining a cluster selector in the following format, `key:value`. See `placementRulePath` to specify an existing file.
+| Specify a placement rule by defining a cluster selector using either `key:value`, or providing a `matchExpressions`, `matchLabels` keys, or both, with appropriate values. See `_placementRulePath_` to specify an existing file.
```
Comment (dhaiducek) [inline governance/policy_generator.adoc]:
```suggestion
| Specify a placement rule by defining a cluster selector using either `key:value`, or providing a `matchExpressions`, `matchLabels`, or both, with appropriate values. See `_placementPath_` to specify an existing file.
```

File: governance/policy_generator.adoc [hunk 11]
```diff
@@ -302,27 +310,39 @@ Note that all the fields in the `policyDefaults` section except for `namespace`
...
 
-| `policies`
-| Required. 
-| The list of policies to create along with overrides to either the default values, or the values that are set in `policyDefaults`. See `policyDefaults` for additional fields and descriptions.
+| *`policySetDefaults`*
+| Optional
+| Any default value for a policy set. Any default value listed for this parameter is overridden by an entry in the `policySets` array.
```
Comment (dhaiducek) [inline governance/policy_generator.adoc]:
```suggestion
| Default values for policy sets. Any default value listed for this parameter is overridden by an entry in the `policySets` array.
```

File: governance/policy_generator.adoc [hunk 12]
```diff
@@ -340,9 +360,9 @@ Note that all the fields in the `policyDefaults` section except for `namespace`
 | Optional
 | A list of Kustomize patches to apply to the manifest at the path. If there are multiple manifests, the patch requires the `apiVersion`, `kind`, `metadata.name`, and `metadata.namespace` (if applicable) fields to be set so Kustomize can identify the manifest that the patch applies to. If there is a single manifest, the `metadata.name` and `metadata.namespace` fields can be patched.
 
-| `policySets`
+| *`policySets`*
 | Optional
-| The list of policy sets to create. To include a policy in a policy set, use `policyDefaults.policySets`, `policies[].policySets`,  or `policySets.policies`.
+| The list of policy sets to create that overrides to either the default values, or the values that are set in `policySetDefaults`. To include a policy in a policy set, use `policyDefaults.policySets`, `policies[].policySets`,  or `policySets.policies`. See `_policySetDefaults_` for additional fields and descriptions.
```
Comment (dhaiducek) [inline governance/policy_generator.adoc]:
```suggestion
| The list of policy sets to create along with overrides to either the default values, or the values that are set in `policySetDefaults`. To include a policy in a policy set, use `policyDefaults.policySets`, `policies[].policySets`,  or `policySets.policies`. See `_policySetDefaults_` for additional fields and descriptions.
```

File: release_notes/whats_new.adoc [hunk 13]
```diff
@@ -47,21 +47,9 @@ For other Application topics, see link:../applications/app_management_overview.a
...
-
-* Use the `toLiteral` function to remove any quotation marks around a template string after it is processed. See link:../governance/custom_template_functions.adoc#toliteral-function[_toLiteral_ function] for more details.
-
-* You can manage policy definitions with OpenShift GitOps (ArgoCD). See link:../governance/manage_policy_def.adoc#manage-policy-definitions[Managing policy definitions with OpenShift GitOps (ArgoCD)].
-
-* You now receive the status events for a policy patching an object in the _History_ column. For more information, see link:../governance/manage_policy_intro.adoc#grc-view[Governance page].
+//clearing what's new for 2.8 | MJ | 03/12/2023
+//2.8 ACM-3011 | MJ | 03/12/2023
+* You can now configure the Policy Generator to add a copy of metadata to replicas of your policies and policy sets, along with generate placement manifests for your policies and policy sets. See link:../governance/policy_generator.adoc#policy-gen-yaml-table[Policy Generator configuration reference table] for more details.
```
Comment (dhaiducek) [inline release_notes/whats_new.adoc]:
```suggestion
* You can now configure the Policy Generator to remove extra metadata when replicating policies, along with specifying placement label selectors for your policies and policy sets. See link:../governance/policy_generator.adoc#policy-gen-yaml-table[Policy Generator configuration reference table] for more details.
```

**Review by dhaiducek — COMMENTED** [review]
A couple wording things. I also realized that I swapped `labelSelector` and `clusterSelector` 🤦‍♂️ , so I've made those suggestions here also.

File: governance/policy_generator.adoc (hunk 12 above)
Comment (dockerymick) [reply governance/policy_generator.adoc]: made an update related to this suggestion.

**Comment by dockerymick** [comment]
@dhaiducek thanks for your suggestions! I accepted most and made a couple more updates.

**Review by dockerymick — APPROVED** [review]
I made a few more changes related to the doc team's effort to reduce the number of inline links. Thanks again for creating this PR, I appreciate you! From my perspective, this PR looks good to me.

**Pull request merged**